online-shop/src/app/details/details.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ActivatedRoute } from '@angular/router';
+import { Post } from '../product/post';
+import { FormBuilder } from '@angular/forms';
+import { Validators } from '@angular/forms';
+import { CartService } from '../cart.service';
+
+@Component({
+  selector: 'app-details',
+  templateUrl: './details.component.html',
+  styleUrls: ['./details.component.css'],
+})
+export class DetailsComponent implements OnInit {
+  product: Post | null = null;
+  idP: number = 0;
+  shoppingCart = new Array<Post>();
+
+  profileForm = this.fb.group({
+    name: [''],
+    category: [''],
+    price: [''],
+    description: [''],
+  });
+
+  constructor(
+    private http: HttpClient,
+    private route: ActivatedRoute,
+    private fb: FormBuilder,
+    private cartService: CartService
+  ) {}
+
+  ngOnInit(): void {
+    const id = this.route.snapshot.paramMap.get('id');
+
+    this.http
```

Comment:
**suggestion:** you need have a Service which handles the logic for retrieving products, for authentication. https://angular.io/guide/architecture-services

Otherwise you are going to repeat yourself by making this requests inside all of your components